### PR TITLE
Implement done command

### LINKED ISSUE #31: implement done command
- Duplicate and Modify UniqueTaskList to support Tasks that are done
- Rename it to be called UniqueDoneList

### LINKED ISSUE #110: [Done Command] There should not be a listdo command
`list` should always list uncompleted task
There shoud not be a `listdo` command
## Suggestion

If want to list both completed and uncompleted tasks, maybe we can have a list-all command, which lists every single tasks in taskmanager.xml file

### PR DESCRIPTION
**Pull request by ZuoZhuolin**
done
list-all
list
list-done

This closes #110 , closes #31

### CODE REVIEW

**Comment by ZuoZhuolin** [comment]
@CS2103AUG2016-W15-C3/code-quality 
@CS2103AUG2016-W15-C3/documentation 
@CS2103AUG2016-W15-C3/testing

File: src/main/java/seedu/taskell/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -40,10 +40,10 @@ public AddCommand(String description, String taskType, String startDate, String
         
         switch (taskType) {
         case Task.FLOATING_TASK: 
-            this.toAdd = new FloatingTask(description, taskPriority, new UniqueTagList(tagSet));
+            this.toAdd = new FloatingTask(description, taskPriority, "incomplete", new UniqueTagList(tagSet));
```
Comment (marcusngwenjian) [inline src/main/java/seedu/taskell/logic/commands/AddCommand.java]: Use the constant in TaskStatus class

File: src/main/java/seedu/taskell/logic/commands/ListDoneCommand.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+public class ListDoneCommand extends Command {
+    
+    public static final String COMMAND_WORD = "list-done";
+
+    public static final String MESSAGE_SUCCESS = "Listed all completed tasks";
+
+    private Set<String> keywordSet = new HashSet<>(Arrays.asList("finished"));
```
Comment (marcusngwenjian) [inline src/main/java/seedu/taskell/logic/commands/ListDoneCommand.java]: Magic string. Use constant in TaskStatus class

File: src/test/java/seedu/taskell/testutil/TestTask.java [hunk 3]
```diff
@@ -16,14 +16,15 @@
...
     private UniqueTagList tags;
 
     public TestTask() {
         tags = new UniqueTagList();
     }
     
     public TestTask(Description description, String taskType, TaskPriority taskPriority, TaskTime startTime, 
-              TaskTime endTime, TaskDate startDate, TaskDate endDate, UniqueTagList tags){
+              TaskTime endTime, TaskDate startDate, TaskDate endDate, TaskStatus taskStatus, UniqueTagList tags){
         this.description = description;
```
Comment (marcusngwenjian) [inline src/test/java/seedu/taskell/testutil/TestTask.java:28]: Whitespace in between brackets

File: src/test/java/seedu/taskell/testutil/TestTask.java [hunk 4]
```diff
@@ -63,10 +65,18 @@ public void setEndDate(TaskDate endDate) {
         this.endDate = endDate;
     }
     
+
+    public void setTaskComplete(TaskStatus taskStatus){
+        this.taskStatus = taskStatus;
```
Comment (marcusngwenjian) [inline src/test/java/seedu/taskell/testutil/TestTask.java:69]: whitespace

**Comment by turtle96** [comment]
testing pass.

for "list-all" command, might be better to change command result "Listed all tasks" to "Listed all incomplete and completed tasks" or something similar

**Pull request merged**